### PR TITLE
Bigtable: 07. Surface - ReadModifyWriteRow

### PR DESCRIPTION
**Pull request by igorbernstein2**
@kevinsi4508, can you take a look when you have a moment?

This the last surface PR

### CODE REVIEW

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 1]
```diff
@@ -333,6 +334,50 @@ public void readRowsAsync(Query query, ResponseObserver<Row> observer) {
...
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * InstanceName instanceName = InstanceName.of("[PROJECT]", "[INSTANCE]");
+   * try (BigtableClient bigtableClient = BigtableClient.create(instanceName)) {
+   *   ReadModifyWriteRow mutation = ReadModifyWriteRow.create("[TABLE]", "[KEY]")
+   *     .increment("[FAMILY]", "[QUALIFIER]", 1)
+   *     .append("[FAMILY2]", "[QUALIFIER2]", "suffix");
+   *
+   *   boolean success = bigtableClient.readModifyWriteRowAsync(mutation);
```
Comment (kevinsi4508) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: This returns a ApiFuture<Row>.

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 2]
```diff
@@ -333,6 +334,50 @@ public void readRowsAsync(Query query, ResponseObserver<Row> observer) {
...
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * InstanceName instanceName = InstanceName.of("[PROJECT]", "[INSTANCE]");
+   * try (BigtableClient bigtableClient = BigtableClient.create(instanceName)) {
+   *   ReadModifyWriteRow mutation = ReadModifyWriteRow.create("[TABLE]", "[KEY]")
+   *     .increment("[FAMILY]", "[QUALIFIER]", 1)
+   *     .append("[FAMILY2]", "[QUALIFIER2]", "suffix");
+   *
+   *   boolean success = bigtableClient.readModifyWriteRowCallable().call(mutation);
```
Comment (kevinsi4508) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: Return type boolean?

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ReadModifyWriteRow.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+  private ReadModifyWriteRow(@Nonnull String tableId, @Nonnull ByteString key) {
+    Preconditions.checkNotNull(tableId, "tableId can't be null.");
+    Preconditions.checkNotNull(key, "key can't be null.");
+    this.tableId = tableId;
+
+    builder.setRowKey(key);
+  }
+
+  public static ReadModifyWriteRow create(@Nonnull String tableId, @Nonnull String key) {
```
Comment (kevinsi4508) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ReadModifyWriteRow.java]: Check null on "key"?

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ReadModifyWriteRow.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    return this;
+  }
+
+  /**
+   * Adds `amount` be added to the existing value. If the targeted cell is unset, it will be treated
+   * as containing a zero. Otherwise, the targeted cell must contain an 8-byte value (interpreted as
+   * a 64-bit big-endian signed integer), or the entire request will fail.
+   */
+  public ReadModifyWriteRow increment(
+      @Nonnull String familyName, @Nonnull String qualifier, long amount) {
```
Comment (kevinsi4508) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ReadModifyWriteRow.java]: Any limitation on 'amount'? can it be 0? negative?

**Comment by garrettjonesgoogle** [comment]
LGTM after @kevinsi4508 's feedback is addressed

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ReadModifyWriteRow.java (hunk 4 above)
Comment (igorbernstein2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ReadModifyWriteRow.java]: I don't know of any limitations, but I think those validations are best left to the server

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ReadModifyWriteRow.java (hunk 3 above)
Comment (igorbernstein2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ReadModifyWriteRow.java]: done

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java (hunk 1 above)
Comment (igorbernstein2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: fixed

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java (hunk 2 above)
Comment (igorbernstein2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: fixed

**Comment by igorbernstein2** [comment]
All feedback should be addressed

**Comment by igorbernstein2** [comment]
Please merge when ready

**Comment by garrettjonesgoogle** [comment]
It looks like you'll need to resolve conflicts first.

**Comment by igorbernstein2** [comment]
Rebased (also squashed to make rebasing easier)

**Comment by igorbernstein2** [comment]
thank you!

**Pull request merged**